### PR TITLE
Hec 1420 rity

### PR DESCRIPTION
**Pull request by priyankadhiman3006**
ticket details -> https://jira.tools.tax.service.gov.uk/browse/HEC-1420

### CODE REVIEW

File: test/uk/gov/hmrc/hecapplicantfrontend/controllers/EntityTypeControllerSpec.scala [hunk 1]
```diff
@@ -350,7 +350,7 @@ class EntityTypeControllerSpec
       "return an InternalServerError" when {
 
         "no selected entity type can be found in session" in {
-          val session = IndividualHECSession(
+          val session = Fixtures.individualHECSession(
             individualLoginData,
             IndividualRetrievedJourneyData.empty,
             IndividualUserAnswers.empty,
```
Comment (meghana-viswanath) [inline test/uk/gov/hmrc/hecapplicantfrontend/controllers/EntityTypeControllerSpec.scala:356]: minor, but we can get rid of the 3 lines below (defaults) if using `Fixtures.relevantIncomeTaxYear` here and in the rest of the file

File: test/uk/gov/hmrc/hecapplicantfrontend/services/JourneyServiceImplSpec.scala [hunk 2]
```diff
@@ -1588,7 +1588,14 @@ class JourneyServiceSpec extends ControllerSpec with SessionSupport {
...
                 val session     =
-                  IndividualHECSession(individualLoginData, journeyData, incompleteAnswers, None, None, List.empty)
+                  Fixtures.individualHECSession(
+                    loginData = individualLoginData,
+                    retrievedJourneyData = journeyData,
+                    userAnswers = incompleteAnswers,
+                    completedTaxCheck = None,
+                    taxCheckStartDateTime = None,
+                    unexpiredTaxChecks = List.empty
+                  )
```
Comment (meghana-viswanath) [inline test/uk/gov/hmrc/hecapplicantfrontend/services/JourneyServiceImplSpec.scala:1598]: same here, we can get rid of the last 3 lines here

File: test/uk/gov/hmrc/hecapplicantfrontend/controllers/EntityTypeControllerSpec.scala (hunk 1 above)
Comment (priyankadhiman3006) [reply test/uk/gov/hmrc/hecapplicantfrontend/controllers/EntityTypeControllerSpec.scala]: updated for this and others [here](https://github.com/hmrc/hec-applicant-frontend/pull/84/commits/87ffce181c080a736c921c28ea47c6c9111db090)

File: test/uk/gov/hmrc/hecapplicantfrontend/services/JourneyServiceImplSpec.scala (hunk 2 above)
Comment (priyankadhiman3006) [reply test/uk/gov/hmrc/hecapplicantfrontend/services/JourneyServiceImplSpec.scala:1595]: [here](https://github.com/hmrc/hec-applicant-frontend/pull/84/commits/87ffce181c080a736c921c28ea47c6c9111db090)

**Review by meghana-viswanath — APPROVED** [review]

File: app/uk/gov/hmrc/hecapplicantfrontend/controllers/TaxSituationController.scala [hunk 3]
```diff
@@ -53,90 +55,99 @@ class TaxSituationController @Inject() (
...
-        individualLoginData: IndividualLoginData,
-        taxSituation: TaxSituation
-      ): EitherT[Future, models.Error, Option[SAStatusResponse]] =
-        if (saTaxSituations.contains(taxSituation)) {
-          individualLoginData.sautr
-            .traverse[EitherT[Future, Error, *], SAStatusResponse](taxCheckService.getSAStatus(_, taxYear))
-        } else {
-          EitherT.pure[Future, models.Error](None)
+      request.sessionData.ensureLicenceTypePresent { licenceType =>
+        val taxYear                                                = individualSession.relevantIncomeTaxYear.getOrElse(getTaxYear(timeProvider.currentDate))
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/hecapplicantfrontend/controllers/TaxSituationController.scala:92]: bu this point we expect there to be a `relevantIncomeTaxYear` in session from the display endpoint - we can safely do a `sys.error` here I think if it's not there. If it's like this in the stride FE we could perhaps change it there as well.

Comment (priyankadhiman3006) [reply app/uk/gov/hmrc/hecapplicantfrontend/controllers/TaxSituationController.scala]: [updated](https://github.com/hmrc/hec-applicant-frontend/pull/84/commits/55611bcfe05feb740755300c62aa6e46e4e1b5ff)

**Review by AndyHWChung — APPROVED** [review]

**Pull request merged**